pkg/operator/etcdcertsigner/etcdcertsignercontroller_test.go

```diff
@@ -0,0 +1,168 @@
+package etcdcertsigner
+
+import (
+	"crypto/rand"
+	"crypto/rsa"
+	"crypto/sha1"
+	"crypto/x509"
+	"crypto/x509/pkix"
+	"fmt"
+	"github.com/openshift/library-go/pkg/crypto"
+	"math/big"
+	"testing"
+	"time"
+)
+
+func Test_verifyCertificateSAN(t *testing.T) {
```

Comment:
(No action required) Given that this is a unit test and you know the implementation, maybe limit the scope of testing to match / doesn't match? `certName` is only used in formulating an error message and the function under test doesn't differentiate between the intended use of the certificate or understand ipv4 vs ipv6 (that's an implementation detail for golang's `VerifyHostname()`).

Reply:
I don't see much point in this unit-test honestly. We are essentially testing VerifyHostname() which is part of the x509 package is already tested by the package (and gazillions users of it)

Reply:
I agree with your assertion that this test adds little value, but the overarching question from my last review remains: How best to ensure that ip address validation is performed correctly and that the intent will be maintained in the face of future maintenance?

Reply:
Yeah. We need to add e2e tests or CI tests with new machines created and IP addresses changed.

Reply:
Can we test kube objects without kube itself? Do we do that already in this repo?

Reply:
@retroflexer As per @hexfusion's comment, please remove this test in favor of one that uses a fake client to simulate API interaction as per [this example](https://github.com/openshift/cluster-etcd-operator/blob/master/pkg/operator/etcdendpointscontroller/etcdendpointscontroller_test.go#L29).